BitList.cs

```diff
@@ -0,0 +1,1496 @@
+        /// Convert this instance to a <see cref="float"/> array.
+        /// </summary>
+        /// <returns>A <see cref="float"/> array.</returns>
+        public float[] ToSingleArray()
+        {
+            if (IsEmpty)
+                return new float[0];
+            else
+            {
+                var ints = ToInt32Array();
+                var floats = new List<float>();
+                foreach (var bits in ints)
+                    floats.Add(BitConverter.Int32BitsToSingle(bits));
+                return floats.ToArray();
+            }
+        }
+
+        /// <summary>
+        /// Convert this instance to a <see cref="double"/> array.
+        /// </summary>
+        /// <returns>A <see cref="double"/> array.</returns>
+        public double[] ToDoubleArray()
+        {
+            if (IsEmpty)
+                return new double[0];
+            else
+            {
+                var longs = ToInt64Array();
+                var doubles = new List<double>();
+                foreach (var bits in longs)
+                    doubles.Add(BitConverter.Int64BitsToDouble(bits));
+                return doubles.ToArray();
+            }
+        }
+
+        /// <summary>
+        /// Convert this instance to a <see cref="decimal"/> array.
+        /// </summary>
+        /// <returns>A <see cref="decimal"/> array.</returns>
+        public decimal[] ToDecimalArray()
+        {
+            if (IsEmpty)
+                return new decimal[0];
+            else
+            {
+                try
+                {
+                    var longs = ToInt64Array().ToList();
+                    if (longs.Count % 2 != 0)
+                        longs.Add(0);
+                    var decimals = new List<decimal>();
+                    for (int i = 0; i < longs.Count; i += 2)
+                        decimals.Add(new decimal(new BitList(new[] { longs[i], longs[i + 1] }).ToInt32Array()));
+                    return decimals.ToArray();
+                }
+                catch
+                {
+                    throw new ConversionErrorException();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Convert this instance to a <see cref="char"/> array.
+        /// </summary>
+        /// <returns>A <see cref="char"/> array.</returns>
+        public char[] ToCharArray()
+        {
+            if (IsEmpty)
+                return new char[0];
+            else
+            {
+                var shorts = ToInt16Array();
+                var chars = new List<char>();
+                foreach (var bits in shorts)
+                    chars.Add((char)bits);
+                return chars.ToArray();
+            }
+        }
+
+        /// <summary>
+        /// Convert this instance to a <see cref="string"/> array.
+        /// </summary>
+        /// <returns>A <see cref="string"/> array.</returns>
+        public override string ToString()
+        {
+            if (IsEmpty)
```

Comment:
Codacy found an issue: [Use the '?:' operator here.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)